thunder/dynamo/utils.py

```diff
@@ -0,0 +1,310 @@
+from enum import Enum, auto
+import dataclasses
+from typing import List, Dict, Optional, Tuple, Set
+from collections.abc import Callable
+import pprint
+import itertools
+import copy
+import inspect
+
+import torch
+from torch.fx._lazy_graph_module import _LazyGraphModule
+import warnings
+from collections.abc import Mapping
+
+from thunder.torch.default_torch_ops import torch_auto_registered_ops
+from thunder.torch import _torch_to_thunder_function_map
+
+
+auto_register_ops = set(itertools.chain(*torch_auto_registered_ops.values()))
+
+
+class CompilerType(Enum):
+    """
+    An enumeration representing different types of compilers.
+    """
+
+    THUNDER = auto()
+    TORCH_INDUCTOR = auto()
+
+
+@dataclasses.dataclass
```

Comment:
would it make sense to make this frozen?
```suggestion
@dataclasses.dataclass(frozen=True)
```